src/lists/LList.es6

```diff
@@ -0,0 +1,197 @@
+import 'core-js/shim';
+let IM = require('immutable');
+
+export default class LList {
+  /**
+   * Accepts items and list-like objects.
+   * Converts them into Immutable Seq's of length >1 before
+   * creating nodes.
+   * @param  {Array}  itemOrList [description]
+   * @param  {Object} options [circular (bool), prependTo(node), oldSize(num)]
+   * @return {[type]}            [description]
+   */
+  constructor(itemOrList = [], options = { circular: false }) {
+    let items = LList.convertToSeq(itemOrList);
+    this.head = LList.makeHead(items);
+    this.size = items.size;
+    let prepend = options.prependTo && LList.isNode(options.prependTo);
+    if (prepend){
```

Comment:
anal, hehe. I say split them -- not nec. to make it more 'pure' but because it will be easier to maintain, I think. :facepunch: